src/MultiplayerMod/Multiplayer/UI/Notifications.cs

```diff
@@ -0,0 +1,27 @@
+using JetBrains.Annotations;
+using MultiplayerMod.Core.Dependency;
+using MultiplayerMod.Core.Events;
+using MultiplayerMod.Multiplayer.CoreOperations.Events;
+
+namespace MultiplayerMod.Multiplayer.UI;
+
+[Dependency, UsedImplicitly]
+public class Notifications {
+
+    public Notifications(EventDispatcher events) {
+        events.Subscribe<ConnectionToHostHasBeenLost>(OnConnectionLost);
+    }
+
+    private void OnConnectionLost(ConnectionToHostHasBeenLost @event) {
+        var screen = (InfoDialogScreen) GameScreenManager.Instance.StartScreen(
+            ScreenPrefabs.Instance.InfoDialogScreen.gameObject,
+            GameScreenManager.Instance.ssOverlayCanvas.gameObject
+        );
+        screen.SetHeader("Connection has been lost");
```

Comment:
Maybe set it to something like "Multiplayer", just to deduplicate it (the same goes in the message).